### DISCUSSION TITLE
Unnecessary reauthentication

### DISCUSSION THREAD

**Discussion by bauche-v** [keycloak/keycloak#26902] [Q&A]
Hello
We have a complete customization for the account management pages, and especialy for the page used to add a new Webauthn key. When I clic on "Add a new key" I am redirected to a custom page served by a custom RealmResourceProvider.
If I use it just after login : no problem, but if I wait 5min I have to reauthenticate...
This reauthentication is asked only for adding a new webauthn, not for adding an OTP application so I guess it's related to my custom provider...
How can I avoid this reauthentication after 5min ?
Here is the log I get :
```
invoke authenticator.authenticate: auth-cookie
authenticator ATTEMPTED: auth-cookie
```

before 5min the log is
```
invoke authenticator.authenticate: auth-cookie
authenticator SUCCESS: auth-cookie
```

keycloak version is 20.0.3
Thanks !

**Comment by danielFesenmeyer** [comment]
Probably you are redirected to a required action. The default `maxAuthAge` is 5 minutes. You could overwrite the corresponding method:
https://github.com/keycloak/keycloak/blob/77581d25278b6819a5cbc802943764557a366c2f/server-spi-private/src/main/java/org/keycloak/authentication/RequiredActionProvider.java#L84

But on the other hand - why do you want to do that? Adding a Webauthn key sounds like something where `maxAuthAge` should not be that long.

**Comment by kmharish** [comment]
Were you able to fix this or find an alternative to this.

I am having the same issue, i am using keycloak to authenticate users to my mobile app, which only uses access token and refresh tokens to keep the uses logged into mobile app for long time.
However I am now trying to provide Password Update and Configure OTP features via required actions, however users are required to reauthenticate (re-login) to do these actions.
